src/main/java/com/edmunds/rest/databricks/service/JobServiceV21.java

```diff
@@ -0,0 +1,234 @@
+   * @see <a href="https://docs.databricks.com/api/latest/jobs.html#delete">https://docs.databricks.com/api/latest/jobs.html#delete</a>
+   * @param jobId The canonical identifier of the job to delete
+   */
+  void deleteJob(long jobId) throws IOException, DatabricksRestException;
+
+  /**
+   * Deletes a job with a given name. Will fail if multiple jobs exist.
+   * @see <a href="https://docs.databricks.com/api/latest/jobs.html#delete">https://docs.databricks.com/api/latest/jobs.html#delete</a>
+   * @param jobName - the job name to delete.
+   */
+  void deleteJob(String jobName) throws IOException, DatabricksRestException;
+
+  /**
+   * Retrieves information about a single job.
+   * @see <a href="https://docs.databricks.com/api/latest/jobs.html#get">https://docs.databricks.com/api/latest/jobs.html#get</a>
+   * @param jobId The canonical identifier of the job to retrieve information about
+   * @return POJO of the Job information
+   */
+  JobDTOv21 getJob(long jobId) throws IOException, DatabricksRestException;
+
+  /**
+   * Retrieves all jobs matching a name exactly.
+   * @see <a href="https://docs.databricks.com/api/latest/jobs.html#list">https://docs.databricks.com/api/latest/jobs.html#list</a>
+   * @param jobName - the full name of the job
+   * @return - all jobs that match.
+   */
+  List<JobDTOv21> getJobsByName(String jobName) throws IOException, DatabricksRestException;
+
+  /**
+   * Will try and get a job by name. If more then one job exists, it will fail. If no job exists it
+   * will return null.
+   * Uses a list behind the scenes, so is O(n) with number of jobs.
+   * @see <a href="https://docs.databricks.com/api/latest/jobs.html#list">https://docs.databricks.com/api/latest/jobs.html#list</a>
+   */
+  JobDTOv21 getJobByName(String jobName) throws IOException, DatabricksRestException;
+
+  /**
+   * Will try and get a job by name. If more then one job exists and failOnMultipleJobs set to
+   * false, it will return the first job created.
+   * Uses a list behind the scenes, so is O(n) with number of jobs.
+   * @see <a href="https://docs.databricks.com/api/latest/jobs.html#list">https://docs.databricks.com/api/latest/jobs.html#list</a>
+   */
+  JobDTOv21 getJobByName(String jobName, boolean failOnMultipleJobs)
+      throws IOException, DatabricksRestException;
+
+
+  /**
+   * Retrieves all jobs with a name matching a given regex.
+   * Uses a list behind the scenes, so is O(n) with number of jobs.
+   * @see <a href="https://docs.databricks.com/api/latest/jobs.html#list">https://docs.databricks.com/api/latest/jobs.html#list</a>
+   * @param regex - the regex to earch for
+   */
+  List<JobDTOv21> getJobsByRegex(Pattern regex) throws IOException, DatabricksRestException;
+
+  /**
+   * Returns a list of all jobs that are active.
+   * @return A POJO of the Jobs
+   */
+  List<JobDTOv21> listAllJobs() throws DatabricksRestException, IOException;
```

Comment:
oh ok you went ahead and kept this method in the new interface?

Reply:
Yes, as it is required for many other methods like :)

Reply:
@samshuster This method is used by `getJobByName`, which is heavily used in tests to delete all test jobs. I had no choice.

Reply:
ah ok yep got it. So the LOE was higher to remove these methods after all.